prov/efa/src/efa_mr.c

```diff
@@ -856,14 +856,16 @@ static int efa_mr_reg_impl(struct efa_mr *efa_mr, uint64_t flags, void *attr)
 	efa_mr->mr_fid.mem_desc = NULL;
 	efa_mr->mr_fid.key = FI_KEY_NOTAVAIL;
 
-	ret = efa_mr_hmem_setup(efa_mr, mr_attr, flags);
+	memcpy(&mr_attr, attr, sizeof(struct fi_mr_attr));
+
```

Comment:
Can we avoid the memcpy by recovering the attr after registering shm mr?

Reply:
AFAICT, if we change *attr to attr in `static int efa_mr_reg_impl(struct efa_mr *efa_mr, uint64_t flags, void *attr)` , we can modify attr inside the function, and it can be recovered after the function call.

Reply:
The reason I changed `mr_attr` from a pointer to a struct is to enforce `efa_mr_reg_impl(..., const void *attr)`, such that `attr` cannot be mutated. In such way we do not need to worry about restoring the value through the test.

wrt `memcpy`, I did not observe any performance difference but, yes, it will add to the overall cost. I'm ok with `memcpy` in this case since MR reg is mostly a one-time procedure. WDYT?

Reply:
>in this case since MR reg is mostly a one-time procedure

this may not be true, Ompi MTL/ofi reg MR every time before it do send, we have mr cache to reduce the cost, but efa_mr_reg_impl can still be called multiple times for buffers that do not overlaps.

Reply:
I see. Then it makes sense to get rid of memcpy. Thanks!